src/Microsoft.Azure.ServiceBus/Core/MessageReceiver.cs

```diff
@@ -160,6 +162,51 @@ internal set
 
         FaultTolerantAmqpObject<RequestResponseAmqpLink> RequestResponseLinkManager { get; }
 
+        private async Task<Message> ProcessMessage(Message message)
+        {
+            if (this.RegisteredPlugins == null)
+            {
+                return message;
+            }
+
+            var processedMessage = message;
+            foreach (var plugin in this.RegisteredPlugins)
+            {
+                try
+                {
+                    MessagingEventSource.Log.PluginCalled(plugin.Name, message.MessageId);
+                    processedMessage = await plugin.AfterMessageReceive(message).ConfigureAwait(false);
+                    MessagingEventSource.Log.PluginCompleted(plugin.Name, message.MessageId);
+                }
+                catch (Exception ex)
+                {
+                    MessagingEventSource.Log.PluginException(plugin.Name, message.MessageId, ex.Message);
+                    if (!plugin.ShouldContinueOnException)
+                    {
+                        throw;
+                    }
+                }
+            }
+            return processedMessage;
+        }
+
+        private async Task<IList<Message>> ProcessMessages(IList<Message> messageList)
+        {
+            if (this.RegisteredPlugins == null || !this.RegisteredPlugins.Any())
```

Comment:
@danielmarbach, looks like this is already corrected in the checked code
https://github.com/Azure/azure-service-bus-dotnet/blob/6cc56bda017d28737b1eae47124cd6d115bc2086/src/Microsoft.Azure.ServiceBus/Core/MessageReceiver.cs#L227